public-ui/src/app/views/app/app.component.ts

```diff
@@ -141,6 +143,37 @@ export class AppComponent implements OnInit {
     head.appendChild(s);
   }
 
+  private setTagManager() {
+    const s = this.doc.createElement('script');
+    s.type = 'text/javascript';
+    s.innerHTML =
+      '(function(w,d,s,l,i){' +
+      'w[l]=w[l]||[];' +
+      'w[l].push({\'gtm.start\':new Date().getTime(),event:\'gtm.js\'});' +
+      'var f=d.getElementsByTagName(s)[0];' +
+      'var j=d.createElement(s);' +
+      'var dl=l!=\'dataLayer\'?\'&l=\'+l:\'\';' +
+      'j.async=true;' +
+      'j.src=\'https://www.googletagmanager.com/gtm.js?id=\'+i+dl+ ' +
+      '\'&gtm_auth=' + environment.gtmAuth + 'gtm_preview=' + environment.gtmPreview +
+      '&gtm_cookies_win=x\';' +
+      'f.parentNode.insertBefore(j,f);' +
+      '})' +
+      '(window, document, \'script\', \'dataLayer\', \'' + environment.gtmId + '\');';
+    const head = this.doc.getElementsByTagName('head')[0];
+    head.appendChild(s);
+  }
+
+  private setTagManagerInBody() {
+    const s = this.doc.createElement('noscript');
```

Comment:
nit: I'd just put these in the same function

Reply:
Done. thanks.